internal/state/change_processor.go

```diff
@@ -199,15 +198,15 @@ func (c *ChangeProcessorImpl) CaptureDeleteChange(resourceType client.Object, ns
 
 func (c *ChangeProcessorImpl) Process(
 	ctx context.Context,
-) (changed bool, conf dataplane.Configuration, statuses Statuses) {
+) (changed bool, conf dataplane.Configuration, g *graph.Graph) {
```

Comment:
I don't think returning full Graph is necessary here. The client doesn't really need it - they care about statuses. Also, dataplane.Configuration can be derived from the Graph, so in a way we're returning duplicate information.

We also started ignoring graph result in the ChangeProcessor tests:
```
changed, conf, _ := processor.Process(context.TODO())
```
which it seems a symptom of the fact that returning the graph isn't neccessary.

What if we return a function that will build statuses. Something like below:


```go
type buildStatusFunc func(reloadErr bool) Statuses

type ChangeProcessor interface {
         . . .
	Process(ctx context.Context) (changed bool, conf dataplane.Configuration, buildStatus buildStatusFunc)
}
```

This way the client will do:
```go

changed, conf, buildStatus := h.cfg.Processor.Process(ctx)
. . .
h.cfg.StatusUpdater.Update(ctx, buildStatus(nginxReloadErr))
```

@kate-osborn  thoughts?

Reply:
Even if we return a function, I wouldn't expect it to be tested in the ChangeProcessor tests, because it would be returning the BuildStatuses function that lives in the `status` package, which we test there.

Reply:
It would abstract the graph, so it doesn't hurt to implement your suggestion

Reply:
@pleshakov @kate-osborn Tried a first pass at cleaning this up. Decided to keep the Process() function because the ChangeProcessor has some state data needed to build the graph that I didn't want to move into the handler.

Reply:
LGTM

Reply:
@sjberman the approach LGTM!

Reply:
Removed a lot of redundant tests and only verify what's necessary for the components under test.